### PR TITLE
Jj profile page

### PR DESCRIPTION
**Pull request by jessjno**
created the username route, set up the show action, added photos under the users and photo commenting

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -22,10 +22,11 @@ def edit
   # POST /comments or /comments.json
   def create
     @comment = Comment.new(comment_params)
+    @comment.author = current_user
 
     respond_to do |format|
       if @comment.save
-        format.html { redirect_to comment_url(@comment), notice: "Comment was successfully created." }
+        format.html { redirect_to fallback_location: root_path, notice: "Comment was successfully created." }
```
Comment (marywebby) [inline app/controllers/comments_controller.rb]: good job implementing the fallback_location, but make sure you are using the correct `redirect` in this situation, double check the lesson at this part [here](https://learn.firstdraft.com/run/84-software-development-foundations-dpi/courses/98-software-development-foundations-dpi/lessons/200-photogram-industrial-part-4#:~:text=seen%20yet%20called-,redirect_back,-%3A)

File: app/controllers/users_controller.rb
Comment (marywebby) [inline app/controllers/users_controller.rb:1]: looking good!!

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -11,22 +11,12 @@
...
-    <%= form.text_field :author_id %>
-  </div>
-
-  <div>
-    <%= form.label :photo_id, style: "display: block" %>
-    <%= form.text_field :photo_id %>
-  </div>
+  <%= form.hidden_field :photo_id %>
 
   <div>
```
Comment (marywebby) [inline app/views/comments/_form.html.erb]: just a little nitpick, but make sure to add the bootstrap inline styling to this div! `class="mb-3"

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <div class="card-body">
+          <h5 class="card-title"><%= photo.owner.username %></h5>
+          <p class="card-text"><%= photo.caption %></p>
+        </div>
+        <ul class="list-group list-group-flush">
+          <% photo.comments.each do |comment| %>
+            <li class="list_group_item">
+              <div class="d-flex">
+                <div class="flex-shrink-0">
+                  <img src="..." alt="...">
```
Comment (marywebby) [inline app/views/users/show.html.erb]: scr and alt are empty, no biggie at the moment, but could be good to place something like `commenter's avatar` or a `placeholder image` in there just for now.

File: config/routes.rb [hunk 4]
```diff
@@ -1,11 +1,14 @@
 Rails.application.routes.draw do
   root "photos#index"
+
   
   devise_for :users
```
Comment (marywebby) [inline config/routes.rb:5]: nice job implementing devise! Apologies what you have is correct.

**Review by marywebby — CHANGES_REQUESTED** [review]
Looking good! You're off to a great start but make sure you go through and make the changes in the places where I have linked back to the lesson. Once you have made those reassign me as a reviewer and I can approve it!

**Review by Testell — APPROVED** [review]
looks good!

**Review by marywebby — APPROVED** [review]
Looks good!!! Thank you for making those changes

**Comment by dgarcia13-99** [comment]
looks good! You did a great job of formatting your files, so everything is nicely indented! Good job making recommended edits!

**Review by dgarcia13-99 — COMMENTED** [review]
looks good! You did a great job formatting your files, so everything is nicely indented! Good job making recommended edits!

**Review by viviancreates — COMMENTED** [review]
Great job refactoring the comment form! Looks like you made some good edits

**Pull request merged**